dts/bindings/gpio/zephyr,gpio-emul.yaml

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020, Friedt Professional Engineering Services, Inc
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO Emulator
+
+compatible: "zephyr,gpio-emul"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+    reg:
+      required: true
+
+    label:
+      required: true
+
+    rising-edge:
+      type: boolean
+
+    falling-edge:
+      type: boolean
+
+    dual-edge:
+      type: boolean
+
+    high-level:
+      type: boolean
+
+    low-level:
+      type: boolean
+
+    dual-level:
```

Comment:
Can't the `duel` properties be defined by having both options (levels or edges) enabled?

Reply:
Not supporting dual but supporting rising / falling or high / low would clearly be very bad hardware design, but for flexibility, I opted to keep them distinct.

Reply:
Maybe I was unclear.  What would it mean to specify that both rising and falling edge detection are supported, but that both edge detection is not supported?  Or that both edge detection is supported, but (say) falling edge detection is not?

By having a distinct property to indicate dual support you open the possibility of inconsistent configurations.  Or am I missing something?

Reply:
Yep, there is the distinct possibility of having inconsistent configurations.

E.g. both rising and falling edge might be unsupported, but dual edge is supported, not that you would necessarily want to do that.

It's flexible enough to support interrupt configuration you can think of, probably even no interrupt capabilities.

Reply:
I don't usually say this because I like flexibility, but allowing support for dual edge/level without allowing both elemental single edge/levels is a level of flexibility I don't think we need: "inconsistent" isn't "flexible".

Please remove the `dual-*` properties.  That's enough to support no interrupts, one sided, other sided, and both sided which together cover the meaningful configurations.

Reply:
OK